### ISSUE TITLE
Update to use ol ESM modules

### ISSUE THREAD

**Issue by geferon** [bjornharrtell/jsts#530]
OpenLayers got updated to ESM modules, and no longer exports everything as a base namespace that can be used for everything.
Even though you can work around this, with the inject functions and etc, it isn't ideal for types and just causes issues with TypeScript.
It would be nice if jsts got a respective update to support the newer versions of `ol`, as right now the only way of using it with TypeScript is adding `skipLibCheck`, which isn't nice imo.

**Comment by bjornharrtell** [comment]
Agreed, even though OpenLayers still offer a bundled script variant. I could probably do something similar as I have in https://github.com/flatgeobuf/flatgeobuf, i.e make ol an optional dependency and have the bundled version of jsts assume that modules exists as globals.

**Comment by bjornharrtell** [comment]
Started work on it at https://github.com/bjornharrtell/jsts/pull/531, considering dropping the single file bundle entirely and major version bump since this change is a breaking one anyway.

**Comment by geferon** [comment]
Perfect! Thank you so much!
